client-data.md

```diff
@@ -27,17 +23,42 @@ The default transfer encoding is Unixfsv1 with the following parameters:
 
 For details on how UnixfsV1 works, see its spec [here](https://github.com/ipfs/specs/tree/master/unixfs).
 
-## Storage Encoding
+## Storage Format
+
+The Storage Format it's the required format for generating Filecoin proofs and hashing sectors data. 
+
+The current Storage Format is `paddedfr32v1`.
+
+### `paddedfr32v1`
+
+A correctly formatted `paddedfr32v1` data must have:
+
+- **Fr32 Padding**: Every 32 bytes blocks must contain two zeroes in the most significant bits (every 254 bits must be followed by 2 zeroes if interpreted as little-endian number). That is, for each block, `0x11000000 & block[31] == 0`.
+- **Piece Padding**: blocks of 32 zero bytes are added so that the total number of blocks (including *piece padding*) is a power of two.
+
+**Why do we need a special Storage Encoding Format?** In the Filecoin proofs we do operations in an arithmetic field of size `p`, where `p` is a prime of size `2^255`, hence the size of the data blocks must be smaller than `p`. We cautiously decide to have data blocks of size 254 to avoid possible overflows (data blocks numerical representation is bigger than `p`). 
+
 
-Storage Encoding is computed by building a merkle tree out of the data using `RepHash`.
 
-Inputs to `RepHash` must first be preprocessed then padded.
+## Piece Commitment
```

Comment:
because it's not necessarely a file, it could be part of a file as you are saying. In addition, there are some proofs called Piece Inclusion Proofs, which would be File Inclusion Proof and we would like to keep FIP for Filecoin Improvement Process